app/Authentication/src/Http/Controllers/MemberSocialAuthController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Authentication\Http\Controllers;
+
+use App\Authentication\Http\Resources\AuthenticatedMemberResource;
+use App\Authentication\Http\Resources\RedirectLinkResource;
+use App\Framework\Enums\RoleName;
+use App\Framework\Models\User;
+use Illuminate\Validation\ValidationException;
+use Knuckles\Scribe\Attributes\Endpoint;
+use Knuckles\Scribe\Attributes\Group;
+use Knuckles\Scribe\Attributes\UrlParam;
+use Laravel\Socialite\Facades\Socialite;
+
+class MemberSocialAuthController
+{
+    #[Group('Authentication')]
+    #[Endpoint(title: 'Socialite Register Redirect', description: 'This endpoint redirect to the Social SignUp Form')]
+    #[UrlParam('provider', 'string', required: true, example: "google", enum: ["github", "google", "linkedin"])]
+    public function redirect($provider)
+    {
+        $this->validateProvider($provider);
+
+        return new RedirectLinkResource(Socialite::driver($provider)->stateless()->redirect()->getTargetUrl());
+    }
+
+    #[Group('Authentication')]
+    #[Endpoint(title: 'Socialite Register Callback', description: 'This endpoint sign up the users with Google, Github or LinkedIn Account')]
+    #[UrlParam('provider', 'string', required: true, example: "google", enum: ["github", "google", "linkedin"])]
+    public function handleCallback($provider)
+    {
+        $this->validateProvider($provider);
+
+        try {
+            $socialiteUser = Socialite::driver($provider)->user();
+            $user = User::where('email', $socialiteUser->getEmail())->first();
+
+            if (!$user) {
+                $user = new User();
+                $user->email = $socialiteUser->getEmail();
+                $user->save();
```

Comment:
Besides saving the email, we will also need the first name, last name and image which should go to the member_profiles table.

Reply:
More info can be found here: #45